Manatee.Json/Pointer/JsonPointer.cs

```diff
@@ -29,16 +29,30 @@ public class JsonPointer : List<string>, IJsonSerializable, IEquatable<JsonPoint
 		/// <param name="source">A collection of strings representing the segments of the pointer.</param>
 		public JsonPointer(params string[] source)
 			: this((IEnumerable<string>) source) { }
+
 		/// <summary>
 		/// Creates a new <see cref="JsonPointer"/> instance.
 		/// </summary>
 		/// <param name="source">A collection of strings representing the segments of the pointer.</param>
 		public JsonPointer(IEnumerable<string> source)
-			: base(source.SkipWhile(s => s == "#"))
+			: base(source.FirstOrDefault() == "#" ? source.SkipWhile(s => s == "#") : source)
```

Comment:
Similarly here with `List<T>` enumerating the input collection.

Reply:
The goal of the change here is to avoid making the SkipWhile call when possible, since there's a cost for spinning up the SkipWhile enumerator.